src/con_duct/__main__.py

```diff
@@ -265,38 +266,50 @@ def __init__(
         self,
         command: str,
         arguments: list[str],
-        session_id: int | None,
-        process: subprocess.Popen,
         log_paths: LogPaths,
         summary_format: str,
         clobber: bool = False,
+        process: subprocess.Popen | None = None,
     ) -> None:
-        self.start_time = time.time()
         self._command = command
         self.arguments = arguments
-        self.session_id = session_id
+        self.log_paths = log_paths
+        self.summary_format: str = summary_format
+        self.clobber = clobber
+        # Defaults to be set later
+        self.start_time = None
+        self.process = process
+        self.session_id: int | None = None
         self.gpus: list[dict[str, str]] | None = None
         self.env: dict[str, str] | None = None
         self.number = 0
         self.system_info: SystemInfo | None = None
-        self.log_paths = log_paths
         self.max_values = Sample()
         self.averages: Averages = Averages()
-        self.process = process
         self.current_sample: Sample | None = None
         self.end_time: float | None = None
         self.run_time_seconds: str | None = None
-        self.summary_format: str = summary_format
-        self.clobber = clobber
 
     @property
     def command(self) -> str:
         return " ".join([self._command] + self.arguments)
 
     @property
     def elapsed_time(self) -> float:
+        assert self.start_time
         return time.time() - self.start_time
 
+    @property
+    def wall_clock_time(self) -> Optional[float]:
+        if self.start_time is None:
+            return math.nan
+        if self.end_time is None:
+            # if no end_time -- must be still ongoing
+            # Cannot happen ATM but could in "library mode" later
```

Comment:
```suggestion
            # Cannot happen ATM but could in "library mode" later
            # TODO: log this
```

Reply:
I misunderstood why this would happen in library mode (this function would be called prior to end), so this isnt a critical/weird thing to happen.